### PR TITLE
feat: refine API error codes

### PR DESCRIPTION
**Pull request by carlcarl**
### Description

Make API errors more precise
This depends on the PR: https://github.com/bnb-chain/zkbnb-crypto/pull/54

### Rationale

Frontend need more details of API error codes

### Example

From general error code
```
20001: invalid param: invalid value for account index
```
To
```
21002: invalid account index
```

### Changes

Notable changes:
* More API error codes

### CODE REVIEW

**Review by forcodedancing — CHANGES_REQUESTED** [review]
for each tx type, it has a function `Validate` to validate the correctness of each field. please also review this part of codes, to see how to handle these validation errors.

**Review by forcodedancing — CHANGES_REQUESTED** [review]
For executors, during `Prepare` (e.g., PrepareAccount), if the resource cannot be found (e.g., account not exists) it will led to `internal error`. We need to review and revise it.

File: core/executor/atomic_match_executor.go [hunk 1]
```diff
@@ -104,10 +104,10 @@ func (e *AtomicMatchExecutor) VerifyInputs(skipGasAmtChk bool) error {
 
 	// Check offer expired time.
 	if err := e.bc.VerifyExpiredAt(txInfo.BuyOffer.ExpiredAt); err != nil {
-		return errors.New("invalid BuyOffer.ExpiredAt")
+		return types.AppErrInvalidBuyOfferExpireTime
```
Comment (forcodedancing) [inline core/executor/atomic_match_executor.go:107]: line 102 needs update. Currently, for atomic match the trading asset should be one of the gas assets.

Comment (carlcarl) [reply core/executor/atomic_match_executor.go:107]: OK, thanks!

**Review by forcodedancing — APPROVED** [review]

**Pull request merged**